### PR TITLE
Clone the vals map for every path to avoid mutation

### PR DESCRIPTION
**Pull request by waveywaves**
**What this PR does / why we need it**:
This PR fixes https://github.com/helm/helm/issues/6079 
`vals` was getting mutated when multiple charts were provided for linting.
Cloning `vals` for every `path` sees to it that every `path` has it's own unmutated set of `vals`.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by waveywaves** [comment]
cc @karuppiah7890 @adamreese @mattfarina

**Comment by karuppiah7890** [comment]
@waveywaves could you add a failing test for the bug and then pass it with the current code? Tests would give confidence on the robustness of the features

**Comment by waveywaves** [comment]
@karuppiah7890 Have updated the PR. Do take a look and let me know if I need to make any changes.

**Comment by karuppiah7890** [comment]
@waveywaves I ran the test without the fix and it passed, may be use existing chart / add some chart that when linted after another one breaks the linting due to the issue? 

For example, in first chart, use a chart with a `config` value which is yaml and has `toYaml` used in the templates to use the `config`, in second chart, use `config` value which is a string and used directly with `.Values.config` with no `toYaml` or anything. The example is a subset of the of the charts mentioned in the issue which caused the bug.

**Comment by karuppiah7890** [comment]
Also, do sign off the commit https://github.com/helm/helm/pull/6241/checks?check_run_id=195950295 for the Developer Certificate of Origin check

**Comment by waveywaves** [comment]
@karuppiah7890 I have added the testcharts against which I was able to have the test fail before the changes. And I have updated the test as well. Let me know if I need to make any changes to this.

**Comment by karuppiah7890** [comment]
I just glanced it. It's awesome 😁. I see you have fixed a Fixme comment too, I have no idea about that, will have to read and understand that. Other than that LGTM. And I can't merge PRs, so cc'ing people. cc @bacongobbler

File: pkg/action/lint.go [hunk 1]
```diff
@@ -59,15 +59,22 @@ func (l *Lint) Run(paths []string, vals map[string]interface{}) *LintResult {
 
 	result := &LintResult{}
 	for _, path := range paths {
-		if linter, err := lintChart(path, vals, l.Namespace, l.Strict); err != nil {
+		linter, err := lintChart(path, vals, l.Namespace, l.Strict)
+		if err != nil {
 			if err == errLintNoChart {
 				result.Errors = append(result.Errors, err)
 			}
+			if linter.HighestSeverity >= lowestTolerance {
```
Comment (thomastaylor312) [inline pkg/action/lint.go]: These changes feel out of scope for this PR because it doesn't seem to be related to the values map mutation. Please let me know if I am wrong

**Review by thomastaylor312 — COMMENTED** [review]
This is looking good. Thanks for adding a good unit test for this! Just one comment around whether a block of changes is needed for this PR

File: pkg/action/lint.go (hunk 1 above)
Comment (waveywaves) [reply pkg/action/lint.go]: None of the `Messages` which are errors are being appended to `Result.Errors` which shouldn't be the case.

Comment (waveywaves) [reply pkg/action/lint.go]: While writing the test, I noticed that there was <nil> getting appended to Result.Errors. This was creating an issue because, Result.Errors was supposed tohave a length of zero, instead it had a length of 1, hence failing my test even on the right code.

When running lintChart on a chart which passes the lint, it would return <nil> for err which is expected. But this lint was being appended to result.Errors which isn't the expected behaviour. Do correct me if I'm wrong.

Comment (thomastaylor312) [reply pkg/action/lint.go]: OK, that make sense then. So this was a different bug you found along the way and fixed also

Comment (thomastaylor312) [reply pkg/action/lint.go]: @waveywaves could you write a unit test to make sure we can validate that in the future?

Comment (waveywaves) [reply pkg/action/lint.go]: @thomastaylor312 I have updated the PR, can you check and let me know.

**Review by thomastaylor312 — APPROVED** [review]
Thanks for the great work on this!
